src/Build/Microsoft.Build.csproj

```
@@ -670,4 +671,30 @@
       <LogicalName>$(AssemblyName).Strings.shared.resources</LogicalName>
     </EmbeddedResource>
   </ItemGroup>
+
+  <!-- Extract binding redirects for Microsoft.Build from MSBuild.exe.config into a source file -->
+  <Target Name="GenerateAppDomainConfig" BeforeTargets="Build" Condition="'$(FeatureAppDomain)' == 'true'">
+    <PropertyGroup>
+      <BindingRedirectNamespace>&lt;Namespace Prefix='ns' Uri='urn:schemas-microsoft-com:asm.v1' /&gt;</BindingRedirectNamespace>
+      <BindingRedirectXPath>/configuration/runtime/ns:assemblyBinding/ns:dependentAssembly[ns:assemblyIdentity/@name='Microsoft.Build']</BindingRedirectXPath>
+    </PropertyGroup>
+
+    <XmlPeek XmlInputPath="..\MSBuild\app.config" Query="$(BindingRedirectXPath)" Namespaces="$(BindingRedirectNamespace)">
+      <Output TaskParameter="Result" ItemName="BindingRedirect32" />
+    </XmlPeek>
+    <XmlPeek XmlInputPath="..\MSBuild\app.amd64.config" Query="$(BindingRedirectXPath)" Namespaces="$(BindingRedirectNamespace)">
+      <Output TaskParameter="Result" ItemName="BindingRedirect64" />
+    </XmlPeek>
+
+    <WriteLinesToFile File="Utilities\NuGetFrameworkWrapper.redirects.cs" Overwrite="true" WriteOnlyWhenDifferent="true" Lines="
```

Is there a reason to prefer to check in the generated file? I would put this in obj

```suggestion
    <WriteLinesToFile File="$(IntermediateOutputPath)NuGetFrameworkWrapper.redirects.cs" Overwrite="true" WriteOnlyWhenDifferent="true" Lines="
```

and add it to `@(Compile)` and make the target incremental. That's the "default pattern" for generated code IMO.